--- a/meson.build	2022-12-05 02:28:21.000000000 +0100
+++ b/meson.build	2023-01-16 13:59:45.659630800 +0100
@@ -17,19 +17,16 @@
 gst_version_is_stable = gst_version_minor.is_even()
 gst_version_is_dev = gst_version_minor.is_odd() and gst_version_micro < 90
 
+cc = meson.get_compiler('c')
+
 api_version = '1.0'
-libavfilter_dep = dependency('libavfilter', version: '>= 7.16.100',
-  fallback: ['FFmpeg', 'libavfilter_dep'])
-libavformat_dep = dependency('libavformat', version: '>= 58.12.100',
-  fallback: ['FFmpeg', 'libavformat_dep'])
-libavcodec_dep = dependency('libavcodec', version: '>= 58.18.100',
-  fallback: ['FFmpeg', 'libavcodec_dep'])
-libavutil_dep = dependency('libavutil', version: '>= 56.14.100',
-  fallback: ['FFmpeg', 'libavutil_dep'])
+libavfilter_dep = declare_dependency(dependencies: [cc.find_library('avfilter')])
+libavformat_dep = declare_dependency(dependencies: [cc.find_library('avformat')])
+libavcodec_dep = declare_dependency(dependencies: [cc.find_library('avcodec')])
+libavutil_dep = declare_dependency(dependencies: [cc.find_library('avutil')])
 
 libav_deps = [libavfilter_dep, libavformat_dep, libavcodec_dep, libavutil_dep]
 
-cc = meson.get_compiler('c')
 static_build = get_option('default_library') == 'static'
 
 check_ffmpeg_src = '''#include <libavcodec/avcodec.h>
